docs/technical-details/reference/globals.md

```diff
@@ -87,13 +87,32 @@ Type: `(request: Request, next: RequestListener) => Promise<Response>`
+
 #### `Request`
+A `Request` object is passed to the direct static REST handlers, and preserved as the context for instance methods, and has the following properties:
+- `url` - This is the request target, which is the portion of the URL that was received by the server. If a client sends a request to `http://example.com:8080/path?query=string`, the actual received request is `GET /path?query=string` and the `url` property will be `/path?query=string`.
+- `method` - This is the HTTP method of the request. This is a string like `GET`, `POST`, `PUT`, `DELETE`, etc.
+- `headers` - This is a [`Headers` object](https://developer.mozilla.org/en-US/docs/Web/API/Headers) that contains the headers of the request.
+- `pathname` - This is the path portion of the URL, without the query string. For example, if the URL is `/path?query=string`, the `pathname` will be `/path`.
+- `protocol` - This is the protocol of the request, like `http` or `https`.
+- `data` - This is the deserialized body of the request (based on the type of data specified by [`Content-Type` header](./content-types.md)).
+- `ip` - This is the remote IP address of the client that made the request (or the remote IP address of the last proxy to connect to Harper).
+- `host` - This is the host of the request, like `example.com`.
+- `sendEarlyHints(link: string, headers?: object): void` - This method sends an early hints response to the client, prior to actually returning a response. This is useful for sending a link header to the client to indicate that another resource should be preloaded. The `headers` argument can be used to send additional headers with the early hints response, in addition to the `link`.
+- `nodeRequest` - This is the underlying Node.js [`http.IncomingMessage`](https://nodejs.org/api/http.html#http_class_http_incomingmessage) object. This can be used to access the raw request data, such as the raw headers, raw body, etc. However, this discouraged and should be used with caution since it will likely break any other server handlers that depends on the layered `Request` call with `Response` return pattern.
+- `nodeResponse` - This is the underlying Node.js [`http.ServerResponse`](https://nodejs.org/api/http.html#http_class_http_serverresponse) object. This can be used to access the raw response data, such as the raw headers. Again, this is discouraged and can cause problems for middleware, should only be used if you are certain that other server handlers will not attempt to return a different `Response` object.
```

Comment:
```suggestion
- `_nodeResponse` - This is the underlying Node.js [`http.ServerResponse`](https://nodejs.org/api/http.html#http_class_http_serverresponse) object. This can be used to access the raw response data, such as the raw headers. Again, this is discouraged and can cause problems for middleware, should only be used if you are certain that other server handlers will not attempt to return a different `Response` object.
```